File: Programming Basics/Programming Basics - JavaScript/Exercises/04. Complex Conditions/19. Hotel Room/HotelRoom.js
function priceForHotelRoom([arg1, arg2]) {
    let month = arg1;
    let nights = Number(arg2);

    let studio;
    let apartment;
    let discountStudio = 1;
    let discountApart = 1;

    if (month == "May" || month == "October") {
        studio = nights * 50;
        apartment = nights * 65;
        if (nights > 7 && nights <= 14) {
            discountStudio *= 0.95;
        }
        else if (nights > 14) {
            discountStudio *= 0.70; 
            discountApart *= 0.90;
        }
    }
    else if (month == "June" || month == "September") {
        studio = nights * 75.20;
        apartment = nights * 68.70;
        if (nights > 14) {
            discountStudio *= 0.80;
            discountApart *= 0.90;
        }
    }
    else if (month == "July" || month == "August") {
        studio = nights * 76;
        apartment = nights * 77;
        if (nights > 14) {
            discountApart *= 0.90;
        }
    }

    console.log(`Apartment: ${(apartment * discountApart).toFixed(2)} lv.`);
    console.log(`Studio: ${(studio * discountStudio).toFixed(2)} lv.`);
}

/* Uncomment to test:
priceForHotelRoom(["May", "15"]);
priceForHotelRoom(["June", "14"]);
priceForHotelRoom(["August", "20"]);
    */

File: Programming Basics/Programming Basics - C#/Exercises/04. Complex Conditionals/04. Complex Conditionals/HotelRoom/HotelRoom.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace HotelRoom
{
    class Program
    {
        static void Main(string[] args)
        {
            string month = Console.ReadLine().ToLower();
            int nights = int.Parse(Console.ReadLine());

            double priceApartment = 0.00;
            double priceStudio = 0.00;

            if (month == "may" || month == "october")
            {
                if (nights <= 7)
                {
                    priceApartment = 65 * nights;
                    priceStudio = 50 * nights;
                }

                else if (nights > 7 && nights <= 14)
                {
                    priceApartment = 65 * nights;
                    priceStudio = (50 - (50 * 0.05)) * nights;
                }

                else if (nights > 14)
                {
                    priceApartment = (65 - (65 * 0.10)) * nights;
                    priceStudio = (50 - (50 * 0.30)) * nights;
                }
            }

            else if (month == "june" || month == "september")
            {
                if (nights <= 14)
                {
                    priceApartment = 68.70 * nights;
                    priceStudio = 75.20 * nights;
                }

                else
                {
                    priceApartment = (68.70 - (68.70 * 0.10)) * nights;
                    priceStudio = (75.20 - (75.20 * 0.20)) * nights;
                }
            }

            else if (month == "july" || month == "august")
            {
                if (nights <= 14)
                {
                    priceApartment = 77 * nights;
                    priceStudio = 76 * nights;
                }

                else
                {
                    priceApartment = (77 - (77 * 0.10)) * nights;
                    priceStudio = 76 * nights;
                }
            }

            Console.WriteLine("Apartment: {0:f2} lv.", priceApartment);
            Console.WriteLine("Studio: {0:f2} lv.", priceStudio);
        }
    }
}
